crates/common/tedge_config/src/tedge_config_cli/figment.rs

```diff
@@ -0,0 +1,325 @@
+use std::borrow::Cow;
+use std::fmt::Display;
+use std::path::Path;
+use std::path::PathBuf;
+
+use figment::providers::Format;
+use figment::providers::Toml;
+use figment::Figment;
+use figment::Metadata;
+use serde::de::DeserializeOwned;
+
+use crate::TEdgeConfigError;
+
+pub trait ConfigSources {
+    fn include_enviroment(&self) -> bool;
```

Comment:
Please fix that typo.

```suggestion
    fn include_environment(&self) -> bool;
```